pkg/proctree/events_processing.go

```diff
@@ -0,0 +1,37 @@
+package proctree
+
+import (
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// ProcessEvent update the process tree according to arriving event
+func (tree *ProcessTree) ProcessEvent(traceeEvent *trace.Event) error {
+	switch traceeEvent.EventName {
```

Comment:
So you are right, using IDs will be better.
The idea @NDStrahilevitz suggested is to put the process tree under the `ProcessEvent` stage in the pipeline, so it will be saved in a map for callbacks there.